### ISSUE TITLE
iOS isn't throwing an AuthError when the user is not confirmed

### ISSUE THREAD

**Issue by ArcezD** [aws-amplify/amplify-flutter#131]
**Describe the bug**
When trying to authenticate using a user with "Account status" "UNCONFIRMED" on iOS the library doesn't throw an Error. While in Android the library throws an AuthError with the details of the Error showing that the user is not confirmed.

**To Reproduce**
1. execute `Amplify.Auth.signUp` to create a new user, but don't call `Amplify.Auth.confirmSignUp`.
2. execute `Amplify.Auth.signIn` using the previously created user.

**Expected behavior**
iOS must have the same behavior as Android throwing an AuthError with the details about the Exception.

**Platform**
[] Android
[x] iOS

**Smartphone (please complete the following information):**
 - Device: iOS Simulator - iPhone 11 Pro
 - OS: iOS 14.0
 - Browser: N/A

**Additional context**
```
Future<String> onLogin(LoginData data) async {
    print('onLogin: Name: ${data.name}, Password: ${data.password}');
    try {
      SignInResult res = await Amplify.Auth.signIn(
        username: data.name.trim(),
        password: data.password.trim(),
      );
      setState(() {
        //isSignedIn = res.isSignedIn;
        print('isSignedIn: ${res.isSignedIn}');
      });
      if (res.nextStep.signInStep == 'DONE') return null;
      if (res.nextStep.signInStep == '') // TODO: Implement go to confirm_sign_up_step
      if (!res.isSignedIn) return 'Ha ocurrido un error';

      return null;
    } on AuthError catch (e) {
      print(e);
      return e.exceptionList
          .firstWhere((i) => i.exception.toLowerCase() == 'platform_exceptions')
          .detail['localizedErrorMessage'];
    }
  }
```

On iOS `Amplify.Auth.signIn` method returns an ﻿`CognitoSignInResult`:
```
{
  "isSignedIn": false,
  "nextStep": {
    "additionalInfo": null,
    "codeDeliveryDetails": {
      "attributeName": "",
      "deliveryMedium": "",
      "destination": ""
    },
    "signInStep": "ERROR"
  }
}
```

On Android `Amplify.Auth.signIn` method returns an ﻿`AuthError`:
```
{
  "cause": "AMPLIFY_SIGNIN_FAILED",
  "exceptionList": [{
      "exception": "USER_NOT_CONFIRMED",
      "detail": "User is not confirmed."
    },
    {
      "exception": "PLATFORM_EXCEPTIONS",
      "detail": {
        "platform": "Android",
        "localizedErrorMessage": "Sign in failed",
        "recoverySuggestion": "See attached exception for more details",
        "errorString": "AmplifyException {message=Sign in failed, cause=com.amazonaws.services.cognitoidentityprovider.model.UserNotConfirmedException: User is not confirmed. (Service: AmazonCognitoIdentityProvider; Status Code: 400; Error Code: UserNotConfirmedException; Request ID: f7031b99-1c40-430c-9745-bdcbce8d3c91), recoverySuggestion=See attached exception for more details}"
      }
    }
  ]
}
```

**Comment by kjones** [comment]
I'm seeing the same thing. Example code with comment about iOS and current temporary workaround:
```dart
  Stream<SignInState> _mapSignInSubmittedToState() async* {
    yield state.withRequestInProgress();
    try {
      final signInResult = await Amplify.Auth.signIn(
          username: state.email, password: state.password);
      if (!signInResult.isSignedIn) {
        // iOS gives zero clues to what happened when signInResult.isSignedIn == false
        // For now, going to assume that when a login fails without throwing an exception
        // that this means "USER_NOT_CONFIRMED" like below.
        yield state.copyWith(mode: SignInMode.confirmSignUp);
      }
    } on AuthError catch (ex) {
      yield (ex.exceptionList
              .any((entry) => entry.exception == "USER_NOT_CONFIRMED"))
          ? state.copyWith(mode: SignInMode.confirmSignUp)
          : state.copyWith(error: ex.cause, exceptions: _toExceptionList(ex));
    } finally {
      yield state.withRequestComplete();
    }
  }
```

**Comment by fjnoyp** [comment]
Hi @ArcezD thanks for pointing out this issue to us,
And to @kjones for the temporary workaround.

We'll discuss this issue internally.

**Comment by Ashish-Nanda** [comment]
@kjones, @ArcezD  Closing this out as a fix for this bug was released today in version `0.0.1-dev.3`. Please give it a try and let us know if you are still facing issues, and we can look into it.

**Comment by BrandowBuenos** [comment]
It looks like the error has returned. There are no exceptions thrown when UserNotConfirmedException error happens.
